polar-core/src/vm.rs

```diff
@@ -790,8 +790,7 @@ impl PolarVirtualMachine {
     }
 
     pub fn source(&self, term: &Term) -> Option<Source> {
-        term.get_source_id()
-            .and_then(|id| self.kb.read().unwrap().sources.get_source(id))
+        self.kb.read().unwrap().get_term_source(term)
```

Comment:
```suggestion
        self.kb().get_term_source(term)
```

Reply:
nice, forgot we recently added that helper